crates/typst-library/src/math/attach.rs

```diff
@@ -84,6 +84,61 @@ impl LayoutMath for AttachElem {
     }
 }
 
+/// Creates specified amount of tight primes
+///
+/// ## Example { #example }
+/// ```example
+/// $ a'''_b = a^'''_b $
+/// ```
+///
+/// ## Syntax
+/// This function has dedicated syntax: use apostrophes instead of primes.
+/// They will automatically attach to previous element,
+/// moving superscripts to the next level.
+///
+/// Display: Attachment
+/// Category: math
+#[element(LayoutMath)]
+pub struct PrimesElem {
+    /// Number of grouped primes
+    #[required]
+    pub count: usize,
+}
+
+impl LayoutMath for PrimesElem {
+    #[tracing::instrument(skip(ctx))]
+    fn layout_math(&self, ctx: &mut MathContext) -> SourceResult<()> {
+        match self.count() {
+            1..=4 => {
+                let f = ctx.layout_fragment(&TextElem::packed(match self.count() {
+                    1 => '′',
+                    2 => '″',
+                    3 => '‴',
+                    4 => '⁗',
+                    _ => unreachable!(),
+                }))?;
+                ctx.push(f)
+            }
+            _ => {
+                // custom amount of primes
+                let prime = ctx.layout_fragment(&TextElem::packed('′'))?.into_frame();
+                let width = prime.width() * (self.count() + 1) as f64 / 2.0;
```

Comment:
why divide by two here? is it an arbitrary measure to make the primes look tight? if so, I wonder (a) how it looks in other fonts and (b) how LaTeX does it.

Reply:
Yeah, without reducing the spacing it will look awful. For unicode symbols with 2/3/4 primes the spacing is larger (the constant `2.0` is smaller, somewhat in `1.5-1.8`, I'm not sure). Nevertheless, it may be useful to reduce the spacing for large number of primes for the better look, and constant `2.0` fits there quite well.

However, in Latex the spacing between primes doesn't change, at least at default font and settings. I thought that in future the spacing for rare case of large prime number could be configured, so I decided not to bother with it now.

Indeed, I should have looked in Latex to understand how they handle it, but going to their source is a thing I wanted to avoid at all costs. :(

I can try, of course.

Reply:
I feel you, looking at how LaTeX does something is always a challenge. The best resource is the TeXBook. I'll leave it up to you, we can also just roll with 2.0.